include/stdexec/__detail/__meta.hpp

```diff
@@ -321,6 +326,9 @@ namespace stdexec {
   template <class... _Args>
   concept _Ok = (STDEXEC_IS_SAME(__ok_t<_Args>, __msuccess) && ...);
 
+  //! If both are true:
+  //! Then __i<true, true>::__g<F, Args...> is an alias for F<Args...>
+  //! and __i<true, true>::__f<F> is an alias for F. 
```

Comment:
I tried to explain what I saw it doing, but I didn't grok it. Is there a high-level description of what `__i` does?

Reply:
it is the implementation of P2300's [_`META-APPLY`_](https://eel.is/c++draft/exec#util.cmplsig-5). 

> [Note [1](https://eel.is/c++draft/exec#util.cmplsig-note-1): The purpose of META-APPLY is to make it valid to use non-variadic templates as Variant and Tuple arguments to gather-signatures. — end note]

in addition to avoiding the dreaded "pack expanded into non-pack argument" error, it is part of the meta-error propagation mechanism. if any of the argument types are a specialization of `_ERROR_`, `__i` will short-circuit and return the error. `__minvoke` and `__meval` are implemented in terms of `__i`.